### PR TITLE
app-editors/bvi: fix strict Clang compilation

### PR DESCRIPTION
**Pull request by Hawk777**
The patch has been sent [upstream](https://sourceforge.net/p/bvi/bugs/14/).

### CODE REVIEW

**Comment by thesamesam** [comment]
Per my message on the ML (https://archives.gentoo.org/gentoo-dev/message/dd9f2d3082b8b6f8dfbccb0639e6e240 but it's 500ing right now...), you do need to revbump for these where they affect code generation.

It was getting the types wrong before. "Build time only fix" really means "when something fixes the build in a way which can't affect runtime behaviour, and we're sure it wasn't miscompiled before"

**Comment by thesamesam** [comment]
Filed https://github.com/gentoo/devmanual/pull/305 too to clarify.

File: app-editors/bvi/files/bvi-1.4.1-prototypes.patch [hunk 1]
```diff
@@ -0,0 +1,32 @@
+diff --git a/bmore.h b/bmore.h
```
Comment (thesamesam) [inline app-editors/bvi/files/bvi-1.4.1-prototypes.patch:1]: Can you include the link to the upstream report here please?

**Comment by Hawk777** [comment]
> Per my message on the ML (https://archives.gentoo.org/gentoo-dev/message/dd9f2d3082b8b6f8dfbccb0639e6e240 but it's 500ing right now...), you do need to revbump for these where they affect code generation.
> 
> It was getting the types wrong before. "Build time only fix" really means "when something fixes the build in a way which can't affect runtime behaviour, and we're sure it wasn't miscompiled before"

Sorry about that; I assumed that because it both compiled and worked fine before then there must not have been problematic codegen. But I suppose there could have been latent failures that weren’t obvious, so I’ve revbumped it.

File: app-editors/bvi/files/bvi-1.4.1-prototypes.patch (hunk 1 above)
Comment (Hawk777) [reply app-editors/bvi/files/bvi-1.4.1-prototypes.patch:3]: Done.

**Comment by thesamesam** [comment]
No problem. In particular, some of these things are particularly hosed on alt arches (but not exclusively).

**Comment by thesamesam** [comment]
Thanks! I also dropped back to ~arch.

**Pull request closed without merging**